src/main/java/org/spongepowered/common/data/processor/value/entity/VanishValueProcessor.java

```diff
@@ -51,16 +52,22 @@ public VanishValueProcessor() {
 
     @Override
     protected boolean set(Entity container, Boolean value) {
-        if (!container.world.isRemote) {
-            EntityUtil.toMixin(container).setVanished(value);
-            return true;
+        if (container instanceof net.minecraft.entity.Entity) {
+            if (!((net.minecraft.entity.Entity) container).world.isRemote) {
+                EntityUtil.toMixin(container).setVanished(value);
+                return true;
+            }
```

Comment:
There is no `return false`, is this working as intended now?

Reply:
No, it's not. Good catch !